src/renderer/components/dock/pod-logs.tsx

```diff
@@ -78,39 +75,81 @@ export class PodLogs extends React.Component<Props> {
     await this.load();
   }
 
+  /**
+   * Function loads more logs (usually after user scrolls to top) and sets proper
+   * scrolling position
+   * @param scrollHeight previous scrollHeight position before adding new lines
+   */
+  preload = async (scrollHeight: number) => {
+    if (podLogsStore.lines < logRange) return;
+    this.preloading = true;
+    await podLogsStore.load(this.tabId).then(() => this.preloading = false);
+    if (this.logsElement.scrollHeight > scrollHeight) {
+      // Set scroll position back to place where preloading started
+      this.logsElement.scrollTop = this.logsElement.scrollHeight - scrollHeight - 48;
+    }
+  }
+
+  /**
+   * Computed prop which returns logs with or without timestamps added to each line and
+   * does separation between new and old logs
+   * @returns {Array} An array with 2 string items - [oldLogs, newLogs]
+   */
   @computed
   get logs() {
-    if (!podLogsStore.logs.has(this.tabId)) return;
-    const { oldLogs, newLogs } = podLogsStore.logs.get(this.tabId);
-    const { getData, removeTimestamps } = podLogsStore;
+    if (!podLogsStore.logs.has(this.tabId)) return [];
+    const logs = podLogsStore.logs.get(this.tabId);
+    const { getData, removeTimestamps, newLogSince } = podLogsStore;
     const { showTimestamps } = getData(this.tabId);
-    return {
-      oldLogs: showTimestamps ? oldLogs : removeTimestamps(oldLogs),
-      newLogs: showTimestamps ? newLogs : removeTimestamps(newLogs)
+    let oldLogs = logs;
+    let newLogs = "";
+    if (newLogSince.has(this.tabId)) {
+      // Finding separator timestamp in logs
+      const index = logs.indexOf(newLogSince.get(this.tabId));
+      if (index !== -1) {
+        // Splitting logs to old and new ones
+        oldLogs = logs.substring(0, index);
+        newLogs = logs.substring(index);
+      }
     }
+    if (!showTimestamps) {
+      return [removeTimestamps(oldLogs), removeTimestamps(newLogs)];
+    }
+    return [oldLogs, newLogs];
   }
 
   toggleTimestamps = () => {
     this.save({ showTimestamps: !this.tabData.showTimestamps });
   }
 
+  /**
+   * Setting 'previous' param to load API request fetching logs from previous container
+   */
   togglePrevious = () => {
     this.save({ previous: !this.tabData.previous });
     this.reload();
   }
 
   onScroll = (evt: React.UIEvent<HTMLDivElement>) => {
```

Comment:
Is there any way to mark some logs as "read" so the "new logs" indicator disappears?

Reply:
Not yet, but this can be added in future along with "clear logs" feature.